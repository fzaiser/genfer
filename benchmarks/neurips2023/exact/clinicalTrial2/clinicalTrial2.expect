Support is a subset of: {0, ..., 1}

Computing moments...
Total measure:             Z = 0.0007359307359307327
Expected value:            E = 0.7549019607843098
2nd raw moment:         μ'_2 = 0.7549019607843098
3rd raw moment:         μ'_3 = 0.7549019607843098
4th raw moment:         μ'_4 = 0.7549019607843098
Standard deviation:        σ = 0.43014531310745924
Variance (2nd central):    V = 0.18502499038831416
3rd central moment:      μ_3 = -0.09432646568815856
4th central moment:      μ_4 = 0.08232224918372683
Skewness (3rd std moment): S = -1.1851899951802083
Kurtosis (4th std moment): K = 2.4046753246752646

Computing probabilities up to 2...
Unnormalized: p(0)     = 0.00018037518037517203
Normalized:   p(0) / Z = 0.24509803921567602
Unnormalized: p(1)     = 0.0005555555555555502
Normalized:   p(1) / Z = 0.7549019607843098
Unnormalized: p(n)     <= 0.0 for all n >= 2
Normalized:   p(n) / Z <= 0.0 for all n >= 2
