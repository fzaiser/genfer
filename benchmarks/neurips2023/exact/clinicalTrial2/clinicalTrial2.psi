def main(){
    isEffective := bernoulli(1/2);
    probIfTreated := uniform(0, 1);
    if isEffective {
        probIfControl := uniform(0, 1);
    } else {
        probIfControl := probIfTreated;
    }

    // Control group:
    observe(0 = bernoulli(probIfControl));
    observe(0 = bernoulli(probIfControl));
    observe(1 = bernoulli(probIfControl));
    observe(0 = bernoulli(probIfControl));
    observe(0 = bernoulli(probIfControl));

    // Treated group:
    observe(1 = bernoulli(probIfTreated));
    observe(0 = bernoulli(probIfTreated));
    observe(1 = bernoulli(probIfTreated));
    observe(1 = bernoulli(probIfTreated));
    observe(1 = bernoulli(probIfTreated));

    return isEffective;
}
