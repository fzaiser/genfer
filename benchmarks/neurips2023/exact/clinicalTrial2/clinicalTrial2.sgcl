isEffective ~ Bernoulli(1/2);
probIfTreated ~ UniformCont(0, 1);
if isEffective = 1 {
    probIfControl ~ UniformCont(0, 1);
} else {
    probIfControl := probIfTreated;
}

# Control group:
observe 0 ~ Bernoulli(probIfControl);
observe 0 ~ Bernoulli(probIfControl);
observe 1 ~ Bernoulli(probIfControl);
observe 0 ~ Bernoulli(probIfControl);
observe 0 ~ Bernoulli(probIfControl);

# Treated group:
observe 1 ~ Bernoulli(probIfTreated);
observe 0 ~ Bernoulli(probIfTreated);
observe 1 ~ Bernoulli(probIfTreated);
observe 1 ~ Bernoulli(probIfTreated);
observe 1 ~ Bernoulli(probIfTreated);

return isEffective;
