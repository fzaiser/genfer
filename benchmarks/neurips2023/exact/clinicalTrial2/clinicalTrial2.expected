77/102
0.75490
